File: Largest_Palindrome_Product/workout.py
from time import time

def iterativeSolution(n):
	start = time()
	isPalindrome = lambda x: x == int(str(x)[::-1])
	upperLimit = (10 ** n) - 1
	lowerLimit = (upperLimit//10) + 1
	maxProduct = 0
	for i in range(upperLimit, lowerLimit - 1, -1):
		for j in range(i, lowerLimit - 1, -1):
			product = i * j
			if product < maxProduct:
				break
			if isPalindrome(product) and product > maxProduct:
				maxProduct = product
	print("Using Loops, Ans : {}, Time : {} seconds".format(maxProduct,time()-start))

N = int(input("N : "))
iterativeSolution(N)

File: Digit_Fifth_Powers/workout.js
const {performance} = require('perf_hooks');
const readline = require('readline').createInterface({
    input: process.stdin,
    output: process.stdout
});

const directSolution = userInput => {
    const START = performance.now();
    
    const N = parseInt(userInput);

    let powers = [];
    for (let i = 0; i <= 9; i += 1)
        powers.push(i ** N);

    const isNarcissistic = number => {
        let numberCopy = number;
        let powerSum = 0;
        while (numberCopy != 0) {
            powerSum += powers[numberCopy % 10];
            numberCopy = Math.floor(numberCopy / 10);
        }
        return powerSum == number;
    };

    const lowerLimit = 10;
    const upperLimit = (N + 1) * (9 ** N);

    let requiredSum = 0;

    for (let no = lowerLimit; no <= upperLimit; no += 1)
        if (isNarcissistic(no))
            requiredSum += no;

    console.log(`Direct Solution, Ans: ${requiredSum}, Time: ${(performance.now()-START)/1000} seconds`);
};

readline.question("N: ", input => {
    directSolution(input);
    readline.close();
});